service/notifier.go

```diff
@@ -106,33 +111,85 @@ func (n *Notifier) prCommentAdded(b *models.WebhookBody) error {
 		}
 
 		// TODO make this domain configurable
-		user, err := n.Slack.GetUserByEmail(mentionedUsername + "@fortnox.se")
+		user, err := slackGetUserByEmail(mentionedUsername + "@fortnox.se")
 		if err != nil {
 			return err
 		}
-		_, _, err = n.Slack.PostMessage(user.ID, b.FormatMessage(b.Comment.Text, "mentioned you in comment")...)
+		_, _, err = slackPostMessage(user.ID, b.FormatMessage(b.Comment.Text, "mentioned you in comment")...)
 		if err != nil {
 			return err
 		}
 	}
 
+	// Notify PR author
 	if b.Actor.Name == b.PullRequest.Author.User.Name { // dont notify yourself
 		logrus.Debug("Skip notifying author is same as actor")
 		return nil
 	}
-	user, err := n.Slack.GetUserByEmail(b.PullRequest.Author.User.EmailAddress)
+	user, err := slackGetUserByEmail(b.PullRequest.Author.User.EmailAddress)
+	if err != nil {
+		return err
+	}
+	_, _, err = slackPostMessage(user.ID, b.FormatMessage(b.Comment.Text, "commented")...)
 	if err != nil {
 		return err
 	}
-	_, _, err = n.Slack.PostMessage(user.ID, b.FormatMessage(b.Comment.Text, "commented")...)
-	return err
-}
 
-func (n *Notifier) getIMByEmail(email string) (string, error) {
-	user, err := n.Slack.GetUserByEmail(email)
+	// Notify comment thread
+	threads := []models.Comment{}
+	activities, err := bitbucketGetPrActivity(b.ID())
 	if err != nil {
-		return "", err
+		return err
+	}
+	for _, v := range activities.Values {
+		if v.Action == "COMMENTED" {
+			threads = append(threads, v.Comment)
+		}
+	}
+	logrus.Debug("Number of comment threads: ", len(threads))
+
+	for _, thread := range threads {
+		authorEmails := map[string]bool{}
+		found := false
+		traverseThread(&thread, authorEmails, b.Comment.ID, &found)
+		if found {
+			delete(authorEmails, b.Comment.Author.EmailAddress)  // never notify the comment author
+			delete(authorEmails, b.PullRequest.Author.User.Name) // never notify the PR author twice (already done before)
+			logrus.Debug("Authors: " + fmt.Sprint(len(authorEmails)))
+
+			for a := range authorEmails {
+				user, err := slackGetUserByEmail(a)
+				if err != nil {
+					return err
+				}
+
+				commentUrl := fmt.Sprintf("%s/projects/%s/repos/%s/pull-requests/%d/overview?commentId=%d/",
+					b.BitbucketURL,
+					b.PullRequest.ToRef.Repository.Project.Key,
+					b.PullRequest.ToRef.Repository.Slug,
+					b.PullRequest.ID,
+					b.Comment.ID,
+				)
+				message := fmt.Sprintf("\n👉 %s\n\n%s", commentUrl, b.Comment.Text)
+				_, _, err = slackPostMessage(user.ID, b.FormatMessage(message, "commented on thread")...)
+				if err != nil {
+					return err
+				}
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func traverseThread(c *models.Comment, authors map[string]bool, commentId int, found *bool) {
+	logrus.Debug("Comment by '" + c.Author.DisplayName + "' | ID: " + fmt.Sprint(c.ID))
+	authors[c.Author.EmailAddress] = true
+	if c.ID == commentId {
+		*found = true
```

Comment:
Do you want to return if found? or can the comments be in other order so we need to go through them all even if the one written is found?